pandas/tests/groupby/test_groupby.py

```diff
@@ -2732,6 +2732,11 @@ def h(df, arg3):
 
         assert_series_equal(result, expected)
 
+    def test_empty_dataframe_groupby(self):
+        # GH8093
+        df = pd.DataFrame(columns=['A', 'B', 'C'])
+        assert df.groupby('A').sum().index.name == 'A'
```

Comment:
Might as well also check that the `DataFrame` that we get as a result is correct.